File: Adm/Models/src/Cliente.java
package src;

import java.util.HashMap;
import java.util.Map;

public class Cliente implements Inter {
  private int id;
  private String nome;
  private String email;
  private String fone;
  private String senha;
  private boolean adm;

  public Cliente(int id, String nome, String email, String fone, String senha, boolean adm) {
    setId(id);
    setNome(nome);
    setEmail(email);
    setFone(fone);
    setSenha(senha);
    setAdm(adm);
  }

  // @Override
  // public String toString() {
  //   return String.format("%d - %s - %s - %s", id, nome, email, fone);
  // }

  public void setId(int id) {
    if (id < 0) {
      throw new IllegalArgumentException("id inválido");
    }
    this.id = id;
  }

  public void setNome(String nome) {
    if (nome == null || nome.isEmpty()) {
      throw new IllegalArgumentException("nome não pode ser vazio");
    }
    this.nome = nome;
  }

  public void setEmail(String email) {
    if (email == null || !email.getClass().equals(String.class)) {
      throw new IllegalArgumentException("Email inválido");
    }
    this.email = email;
  }

  public void setFone(String fone) {
    if (fone == null || fone.isEmpty()) {
      throw new IllegalArgumentException("fone não pode ser vazio");
    }
    this.fone = fone;
  }

  public void setSenha(String senha) {
    if (senha == null || senha.isEmpty()) {
      throw new IllegalArgumentException("senha não pode ser vazia");
    }
    this.senha = senha;
  }

  public void setAdm(boolean adm) {
    this.adm = adm;
  }

  public int getId() {
    return id;
  }

  public String getNome() {
    return nome;
  }

  public String getEmail() {
    return email;
  }

  public String getFone() {
    return fone;
  }

  public String getSenha() {
    return senha;
  }

  public boolean isAdm() {
    return adm;
  }

  public Map<String, Object> toDict() {
    Map<String, Object> dict = new HashMap<>();
    dict.put("id", this.id);
    dict.put("nome", this.nome);
    dict.put("email", this.email);
    dict.put("fone", this.fone);
    dict.put("senha",this.senha);
    dict.put("adm",this.senha);
    return dict;
}
  @Override
  public String toString() {
      return "Item{" +
              "id=" + id +
              ", nome=" + nome +
              ", email=" + email +
              ", fone=" + fone +
              ", senha=" + senha +
              "adm" + adm +
              '}';
  }
}

File: Adm/Models/Cliente.cs
using System;
using System.Collections.Generic;

namespace Adm.Models
{
    public class Cliente : Inter
    {
        // Atributos privados
        private int id;
        private string nome;
        private string email;
        private string fone;
        private string senha;
        private bool adm;

        // Construtor que apenas inicializa os campos, sem validação
        public Cliente(int id, string nome, string email, string fone, string senha, bool adm)
        {
         setId(id);   // Usando os setters para garantir a validação
         setNome(nome);
         setEmail(email);
         setFone(fone);
         setSenha(senha);
         setAdm(adm);
        }

        // Métodos setters com validação
        public void setId(int id)
        {
            if (id < 0)
            {
                throw new ArgumentException("ID inválido");
            }
            this.id = id;
        }

        public void setNome(string nome)
        {
            if (string.IsNullOrWhiteSpace(nome))
            {
                throw new ArgumentException("Nome não pode ser vazio");
            }
            this.nome = nome;
        }

        public void setEmail(string email)
        {
            if (string.IsNullOrWhiteSpace(email))
            {
                throw new ArgumentException("Email inválido");
            }
            this.email = email;
        }

        public void setFone(string fone)
        {
            if (string.IsNullOrWhiteSpace(fone))
            {
                throw new ArgumentException("Fone não pode ser vazio");
            }
            this.fone = fone;
        }

        public void setSenha(string senha)
        {
            if (string.IsNullOrWhiteSpace(senha))
            {
                throw new ArgumentException("Senha não pode ser vazia");
            }
            this.senha = senha;
        }

        public void setAdm(bool adm)
        {
            this.adm = adm;
        }

        // Métodos getters
        public int getId() => this.id;
        public string getNome() => this.nome;
        public string getEmail() => this.email;
        public string getFone() => this.fone;
        public string getSenha() => this.senha;
        public bool isAdm() => this.adm;

        // Método para converter cliente em dicionário
        public Dictionary<string, object> ToDict()
        {
            return new Dictionary<string, object>
            {
                { "id", this.id },
                { "nome", this.nome },
                { "email", this.email },
                { "fone", this.fone },
                { "senha", this.senha },
                { "adm", this.adm }
            };
        }

        // Método ToString
        public override string ToString()
        {
            return $"Cliente: id={id}, nome={nome}, email={email}, fone={fone}, senha={senha}, adm={adm}";
        }
    }
}
